extensions/database-documentation/src/common/utils.ts

```diff
@@ -0,0 +1,179 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as azdata from 'azdata'
+import * as vscode from 'vscode';
+// Language localization features
+import * as nls from 'vscode-nls';
+// The module 'openai' contains the OpenAI API
+import { Configuration, OpenAIApi } from "openai";
+
+const localize = nls.loadMessageBundle();
+
+export async function generateMarkdown(context: azdata.ObjectExplorerContext): Promise<string> {
+	let connection = (await azdata.connection.getCurrentConnection());
+	let databaseName = context.connectionProfile.databaseName;
+
+	let tables: string[];
+	if (context.nodeInfo.nodeType === 'Table') {
+		tables = [context.nodeInfo.metadata.name];
+	}
+	else {
+		const queryProvider = azdata.dataprotocol.getProvider<azdata.QueryProvider>("MSSQL", azdata.DataProviderType.QueryProvider);
+		const connectionUri = await azdata.connection.getUriForConnection(connection.connectionId);
+		let query = `SELECT TABLE_NAME FROM [${databaseName}].INFORMATION_SCHEMA.TABLES WHERE TABLE_TYPE = 'BASE TABLE'`;
+
+		if (context.nodeInfo.nodeType === 'Schema') {
+			query += ` AND TABLE_SCHEMA = '${context.nodeInfo.metadata.name}'`;
+		}
+
+		tables = (await queryProvider.runQueryAndReturn(connectionUri, query)).rows.map(row => row[0].displayValue);
+	}
+
+	let diagram = '```mermaid\nclassDiagram\n';
```

Comment:
have you tried using an erDiagram?  I think that is the more appropriate chart for a db schema and wonder if there is any differences built in, such as displaying cardinalities on relationships, or something like that?